packages/components/src/navigator/navigator-back-link/component.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { contextConnect, WordPressComponentProps } from '../../ui/context';
+import { View } from '../../view';
+import { useNavigatorBackLink } from './hook';
+import type { NavigatorBackLinkProps } from '../types';
+
+function NavigatorBackLink(
+	props: WordPressComponentProps< NavigatorBackLinkProps, 'button' >,
+	forwardedRef: Ref< any >
+) {
+	const navigatorBackLinkProps = useNavigatorBackLink( props );
+
+	return <View ref={ forwardedRef } { ...navigatorBackLinkProps } />;
+}
+
+/**
+ * The `NavigatorBackLink` component can be used to navigate to a screen and
+ * should be used in combination with the `NavigatorProvider`, the
+ * `NavigatorScreen` and the `NavigatorLink` components (or the `useNavigator`
+ * hook).
+ *
+ * @example
+ * ```jsx
+ * import {
+ *   __experimentalNavigatorProvider as NavigatorProvider,
+ *   __experimentalNavigatorScreen as NavigatorScreen,
+ *   __experimentalNavigatorLink as NavigatorLink,
+ *   __experimentalNavigatorBackLink as NavigatorBackLink,
+ * } from '@wordpress/components';
+ *
+ * const MyNavigation = () => (
+ *   <NavigatorProvider initialPath="/">
+ *     <NavigatorScreen path="/">
+ *       <p>This is the home screen.</p>
+ *        <NavigatorLink variant="secondary" path="/child">
```

Comment:
> If the intent is to give it secondary styling by default, the NavigatorLink component should perhaps handle that internally.

This was partially the intent. But your point makes sense — I went ahead and removed `variant="secondary"` from the examples in https://github.com/WordPress/gutenberg/pull/38634/commits/68fac76e6b660d37b2ad9da248b0aaae57b44796.

Also wanted to note that we're using that same prop in the Storybook examples

> Looking at the actual use cases in our codebase though, I think we can assume that most consumers will be using this via as prop or hook. Should we maybe highlight that kind of example instead? If this is the main use case, we could even simply further and make the internal component a lowercase-b <button>.

Not not about a lot of consumers using the `as` prop — I think that using `Button` is a sensible default that may cover a good amount of scenarios. For more context, this was discussed originally in https://github.com/WordPress/gutenberg/pull/37223#discussion_r769391011


> If we do want to highlight that it's internally a Button, that should probably be made clearer in the docs.

Related to https://github.com/WordPress/gutenberg/pull/37223#discussion_r769391011, actually there's an interesting point that caught my attention:

_I believe that `Button`, when passed an `href` prop, automatically renders as a `<a />` instead of a `<button />`_

In the light of the above, should we disallow `href` as a prop from `NavigatorLink`? I could then go ahead and add an additional paragraph at the end of the `Props` section, similarly to https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/card/card/README.md?plain=1#L67-L69

Reply:
> _I believe that `Button`, when passed an `href` prop, automatically renders as a `<a />` instead of a `<button />`_
> 
> In the light of the above, should we disallow `href` as a prop from `NavigatorLink`? I could then go ahead and add an additional paragraph at the end of the `Props` section, similarly to [https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/card/card/README.md?plain=1#L67-L69](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/card/card/README.md?plain=1&rgh-link-date=2022-02-09T21%3A16%3A38Z#L67-L69)

Ah, that's right. Let's do that.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/38634/commits/5c584a5dd18031e7c3d47f96286b7ce197b9cb54